language/benchmarks/src/move_vm.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use bytecode_verifier::VerifiedModule;
+use criterion::Criterion;
+use libra_state_view::StateView;
+use libra_types::access_path::AccessPath;
+use libra_types::account_address::AccountAddress;
+use libra_types::identifier::{IdentStr, Identifier};
+use libra_types::language_storage::ModuleId;
+use move_lang::shared::Address;
+use move_lang::to_bytecode::translate::CompiledUnit;
+use vm::{
+    gas_schedule::{CostTable, GasAlgebra, GasUnits},
+    transaction_metadata::TransactionMetadata,
+};
+use vm_runtime::{
+    chain_state::TransactionExecutionContext, data_cache::BlockDataCache, move_vm::MoveVM,
+};
+
+/// Entry point for the bench, provide a function name to invoke in Module Bench in bench.move.
+pub fn bench(c: &mut Criterion, fun: &str) {
+    let module = compile_module();
+    let mut move_vm = MoveVM::new();
+    move_vm.cache_module(module);
+    execute(c, &move_vm, fun);
+}
+
+// Compile `bench.move`
+fn compile_module() -> VerifiedModule {
+    // TODO: this has only been tried with `cargo bench` from `libra/src/language/benchmarks`
```

Comment:
Can you just `include_str!` the `.move` file? That will allow it to be used from anywhere.

Reply:
I am not sure what you are suggesting here and if it matters.
Could you clarify?

Reply:
So right now you look for the file from the current directory, which can break if you run the benchmark from a different directory. You can use `CARGO_MANIFEST_DIR` instead of the current dir.

See https://github.com/libra/libra/blob/f86d084141bd40b4335f7c09b9d66cb18f831116/language/e2e-tests/src/data_store.rs#L22 for an example.